docs/computing/connecting.md

```diff
@@ -1,93 +1,122 @@
+If you for some reason want to use a slower, X11 based graphical connection, your local computer
+must have an X server program installed and running. In Linux and macOS an X server is normally
+installed automatically, while for Windows it needs to be installed separately. A free X server
+for Windows is provided, for example, by [MobaXterm](https://mobaxterm.mobatek.net/) or
+[Xming](http://www.straightrunning.com/XmingNotes/).
 
-Depending on your local `ssh` version, you may also need to add option `-X` or `-Y` to your ssh command:
+Depending on your local `ssh` version, you may also need to add option `-X` or `-Y` to
+your ssh command:
 
-```
+```bash
 ssh -X yourcscusername@puhti.csc.fi
 ```
 
-In `PuTTY`, X11 forwarding is enabled in the connection settings (Connection -> SSH -> X11: Enable X11 forwarding).
+In `PuTTY`, X11 forwarding is enabled in the connection settings (Connection -> SSH
+-> X11: Enable X11 forwarding).
 
 ## Setting up SSH keys
 
-SSH keys provide more secure authentication, which can be enabled with a two-step process:
+SSH keys provide more secure authentication and can be enabled with a two-step process:
 
-1. **Generate SSH Keys on your local PC**<br> 
-   The SSH Keys are always generated in pairs, one *public key* and
-   one *private key*. These keys should be generated on the computer
-   you are using to connect to CSC supercomputers.    
-2. **Copy public key from local PC to supercomputer**<br>
-	Only the *public key* should be copied, don't copy the private key. 
+1. **Generate SSH Keys on your local PC**  
+   The SSH Keys are always generated in pairs, one _public key_ and
+   one _private key_. These keys should be generated on the computer
+   you are using to connect to CSC supercomputers.
+2. **Copy public key from local PC to supercomputer**  
+   Only the _public key_ should be copied, don't copy the private key.
 
 !!! warning "Note"
-    The private key should never be shared with anyone, not even with
-    CSC staff. It should be also stored only in the local computer (Public key
+    The private key should **never** be shared with anyone, not even with
+    CSC staff. It should be also stored only on the local computer (public key
     can be safely stored in cloud services).
 
-An SSH key pair can be generated in the Linux, macOS, Windows PowerShell and MobaXterm terminal as follows. For Putty, see [PuTTy SSH keys instructions](#ssh-keys-with-putty) below.
+An SSH key pair can be generated in the Linux, macOS, Windows PowerShell and MobaXterm
+terminals as follows. For PuTTY, see [PuTTY SSH keys instructions](#ssh-keys-with-putty)
+below.
 
 ```bash
-ssh-keygen -t rsa -b 4096
+ssh-keygen -o -a 100 -t ed25519
 ```
 
 You will be prompted for a file name and location where to save the
-key. Accept the defaults by pressing ENTER.
+key. Accept the defaults by pressing `ENTER`.
```

Comment:
We might add a note here that `ssh-keygen` will warn the user if the default file name would overwrite an existing key, in which case the user very like doesn't want to accept that. Then it's better to give a different file name manually.

Reply:
It will look like this, in case you want to make a clear example:
```
$ ssh-keygen -o -a 100 -t ed25519
Generating public/private ed25519 key pair.
Enter file in which to save the key (/home/username/.ssh/id_ed25519): 
/home/username/.ssh/id_ed25519 already exists.
Overwrite (y/n)? n
```

Reply:
Then instead make a more explicit key file name:
```
$ ssh-keygen -o -a 100 -t ed25519
Generating public/private ed25519 key pair.
Enter file in which to save the key (/home/username/.ssh/id_ed25519): /home/username/.ssh/id_ed25519-puhti
Enter passphrase (empty for no passphrase): 
Enter same passphrase again: 
Your identification has been saved in /home/username/.ssh/id_ed25519-puhti.
Your public key has been saved in /home/username/.ssh/id_ed25519-puhti.pub.
The key fingerprint is:
SHA256:7yvhmPa/kdpOh1tWDYw/uGmdyvwuPq9WRg0nFBmnavA username@hostname
The key's randomart image is:
+--[ED25519 256]--+
|             .++.|
|             o+o.|
|          . . += |
|           o +.o.|
|        S   E.+ .|
|        .. + =oo |
|       + .* Boo  |
|      + o= X+.   |
|     . .o+O=*B+  |
+----[SHA256]-----+
```

It would also be useful to link to the section about "SSH key file with non-default name or location" which shows up later.

Reply:
Thanks for the suggestion, I'll add this!